PLANNER-1958: Allow exporting of Shift Roster to Excel

Christopher-Chianelli opened the pull request:
Current format is:
Start, End, Employee

with the following caveats to maximize human readability:

- Shifts with the same date and time in the same spot don't have their date/time repeated
- Employee(s) are 1 line under date/time

Ex:

Start,       End,           Employee
date 1,     date 2,
                                  Amy
                                  Bob
date 3,     date 4,
                                  Charlie

yurloc commented on optaweb-employee-rostering-frontend/src/ui/pages/shift/ExportScheduleModal.tsx:
```diff
@@ -0,0 +1,127 @@
+import moment from 'moment';
+
+interface StateProps {
+    tenantId: number;
```
This file is breaking many code style rules. Please follow the ESLint rules that we have configured to ensure consistent code style.

Christopher-Chianelli commented:
@yurloc fixed your issues; the data being reset when solving was due to shallow equality (and among other pitfalls, Dates are not shallowly equal if they refer to the same date). I fix the shallow equality issue by making selectors return the same list if the idMap did not change since the last call, and using the getTime method on the dates. Although I did disable the export button if scheduling is in process, as it doesn't make sense to export the data while it is changing, but I should prevent the issue in other components and reduce rerendering/recomputation.
